src/com/company/Main.java

```diff
@@ -133,113 +172,188 @@ public static void _readBookList() {
         library.traverseBook(library.rootBook, listOfBooks);
-        List<BookNode> sortedUsers = listOfBooks.stream().filter((book)-> book.availableCount == 0).collect(Collectors.toList());
-        writeToOutputFile(sortedUsers, prompt);
+        List<BookNode> sortedUsers = listOfBooks.stream().filter((book) -> book.availableCount == 0).collect(Collectors.toList());
+        try {
+            initializeWriter();
+            if(sortedUsers.isEmpty()){
+                System.out.println("No book in the Library is out of stock");
+            }
+            else {
+                writer.write("All available copies of the below books have been checked out:\n");
+                for (BookNode book : sortedUsers) {
+                    writer.write(""+book.bookId+"\n");
+                }
+            }
+            writer.write("\n");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
     }
 
     //Operation 7
-    public static void _printBooks(String prompt) {
-        List<BookNode> listOfBooks= new ArrayList<BookNode> ();
+    public static void _printBooks() {
+        List<BookNode> listOfBooks = new ArrayList<BookNode>();
         library.traverseBook(library.rootBook, listOfBooks);
-        writeToOutputFile(listOfBooks,prompt);
-    }
-
-
-    public static void writeToOutputFile(List<BookNode> list, String prompt){ // printing list content to outputPS4.txt
         try {
-            BufferedWriter writer = new BufferedWriter(new FileWriter("outputPS4.txt"));
-            for(BookNode book:list) {  // printing data of each book
-                writer.write("Book Id: "+book.bookId+", AvailableCounter: "+book.availableCount+", CheckoutCounter: "+book.checkoutCounter+"\n");
+            initializeWriter();
+            if(listOfBooks.isEmpty()){
+                System.out.println("There are currently no books in the Library");
+            }
+            else {
+                writer.write("There are a total of "+listOfBooks.size()+" book titles in the library.\n");
+                for (BookNode book : listOfBooks) {
+                    writer.write(""+book.bookId+", "+book.availableCount+"\n");
+                }
             }
             writer.write("\n");
-            writer.flush();
-            writer.close();
-        } catch (IOException e){ // error handling while writing to file
+        } catch (IOException e) {
             e.printStackTrace();
         }
     }
 
-    public static void triggerFunctionforPrompts(List<String> prompts) {
-        for(String prompt:prompts) {
-            if (prompt.equalsIgnoreCase("ListTopBooks")) {
-                _getTopBooks(prompt);
-            }
-            else if(prompt.equalsIgnoreCase("BooksNotIssued")) {
-                _notIssued(prompt);
-            }
-            else if(prompt.equalsIgnoreCase("ListStockOut")) {
-                _stockOut(prompt);
+    //operation 8
+    public static void _findBook(int bkID) {
```

Comment:
check operation numbers for all the functions and correct it as per pdf. findbook is operation 5 and not 7